BPA/Ch.10/js_super_indran.stan

```diff
@@ -104,98 +182,40 @@ transformed parameters {
   // Constraints
   for (i in 1:M) {
     for (t in 1:(n_occasions - 1))
-      phi[i, t] <- mean_phi;
+      phi[i, t] = mean_phi;
     for (t in 1:n_occasions)
-      p[i, t] <- inv_logit(logit(mean_p) + epsilon[i]);
-  } //i
+      p[i, t] = inv_logit(logit(mean_p) + epsilon[i]);
+  }
 
   // Dirichlet prior for entry probabilities
   // beta ~ gamma(1, 1);  // => model block
-  b <- beta / sum(beta);
+  b = beta / sum(beta);
 
   // Convert entry probs to conditional entry probs
-  nu[1] <- b[1];
+  nu[1] = b[1];
   for (t in 2:(n_occasions - 1))
-    nu[t] <- b[t] / (1.0 - sum(b[1:(t - 1)]));
-  nu[n_occasions] <- 1.0;
+    nu[t] = b[t] / (1.0 - sum(b[1:(t - 1)]));
+  nu[n_occasions] = 1.0;
 
   // Uncaptured probability
-  chi <- prob_uncaptured(M, n_occasions, p, phi);
+  chi = prob_uncaptured(M, n_occasions, p, phi);
 }
 
 model {
-  vector[n_occasions] qnu;
-
-  qnu <- 1.0 - nu;
-
   // Priors
-  // An improper flat prior (0,+infinity) is implicitly used on sigma.
-  mean_phi ~ uniform(0, 1);
-  mean_p ~ uniform(0, 1);
+  // Uniform priors are implicitly defined.
+  //  mean_phi ~ uniform(0, 1);
+  //  mean_p ~ uniform(0, 1);
+  //  psi ~ uniform(0, 1);
+  //  sigma ~ uniform(0, 5);
+  // In case of using a weakly informative prior
```

Comment:
The above comments are fine, because they are bout what the code does.  The following comment about weakly informative priors opens a can of worms.  When you start making comments like this, where do you stop?  Probably better not to include this in the code. Maybe you could write a paper or monograph on how to code all these models in Stan and provide advice there?